src/libraries/System.Private.CoreLib/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -72,7 +73,18 @@ public Dictionary(int capacity, IEqualityComparer<TKey>? comparer)
                 if (typeof(TKey) == typeof(string) &&
                     NonRandomizedStringEqualityComparer.GetStringComparer(_comparer!) is IEqualityComparer<string> stringComparer)
                 {
+#if TARGET_BROWSER
```

Comment:
It sounds pretty broken to be changing behavior of generic collections when hybrid globalization is enabled. Can this be fixed in the string comparers instead, so that anybody using the string comparers gets the right behavior?

Reply:
I like it

Reply:
I like the idea but so far the implementation of it that I have in my head is also messy. So we could go to https://github.com/dotnet/runtime/blob/9e31c21bcbb661fc4fa235839a66442a65ef447c/src/libraries/System.Private.CoreLib/src/System/StringComparer.cs#L15
and for each comparer option (`InvariantCulture`, `InvariantCultureIgnoreCase`, etc) make a preprocessor directive `#if TARGET_BROWSER` then use `Ordinal` or `OrdinalIgnoreCase` instead. However, it is also problematic:
- All the usages of these `StringComparer` types, even in the context of collections that are currently not affected by `PNSE` will be changed.
- User that calls `StringComparer.CurrentCulture` expects current culture, not `Ordinal`. This can be documented but does not look good.
Or maybe I missed the point and you had a different idea, @jkotas

Reply:
My understanding we are playing tradeoffs here. Hybrid is partial between ICU and InvariantCulture.

`StringComparer.CurrentCulture` expects locale, but we can't do it because we don't have hybrid hashcode, right ?
So we could keep PNSE or we could do `InvariantCulture`, that's our tradeoff.

Or is `Ordinal` better ? Why ?

Reply:
Because it does not use ICU by default so we don't need native `SortKey` implementation to keep it working as expected. The idea was: only `Ordinal` and `OrdinalIgnoreCase` are able to work in `Hybrid Globalization`, so use them always for collections that wanted to use other options and thanks to that, keep these collections supported. If we "keep PNSE or we could do InvariantCulture" (`InvariantCulture` throws the same way as any other culture, because it uses `CompareInfo` implementation) then we would just stick to the current behavior. And we don't want it, we would like to make using collections listed in the description of this PR available to the users.

Reply:
I believe you're mixing `InvariantCulture` and `InvariantGlobalization`. Hybrid is partial between ICU and InvariantCulture. -> "Hybrid is partial between ICU and InvariantGlobalization.

When we use `InvariantCulture` we still use the same paths in the code as for `ICU`.

Reply:
There are hundreds of different types of custom collections out there in the ecosystem, and many of them are affected the same way as the few collections that you are changing in this PR. It is not ok to break all these custom collections out there with hybrid globalization. I would be a major violation of our code portability and compatibility promise.

Reply:
@jkotas do you mean the other solution that I proposed is also invalid? If so, I am really out of ideas

Reply:
Due to limitations of Web API the Hybrid Globalization feature in the Browser in fact turned out to be less promising than e.g. iOS's implementation where we were able to really maintain ICU-like experience with Invariant-like size. However, the unsupported scope is not that huge, we are listing all the limitations in a doc (https://github.com/dotnet/runtime/blob/928ff3015d1936ca9985a0123754e13cbf47b237/docs/design/features/globalization-hybrid-mode.md). I do not believe a re-design would help in the issue we have with `SortKey`. We know we cannot support it nativity (and loading it from ICU does not pay off in this point, we would loose all the benefits of collations removal). Now we have a brainstorm how to solve it some other way.